lib/PrintAsClang/DeclAndTypePrinter.cpp

```diff
@@ -2999,6 +3006,14 @@ bool DeclAndTypePrinter::shouldInclude(const ValueDecl *VD) {
       return false;
   }
 
+  // In C output mode print only @cdecls and skip them in other modes.
+  bool isCDeclForC = false;
+  auto *FD = dyn_cast<AbstractFunctionDecl>(VD);
```

Comment:
Nit: could move `FD` to the if condition.

Reply:
Makes sense! It's updated.